### ISSUE TITLE
Drag-and-drop file upload UI

### ISSUE THREAD

**Issue by kaganjd** [processing/p5.js-web-editor#375]
Based on #92 and #271, @machinic designed a new UI for file uploads! @machinic, @catarak, @lmccart: I'm guessing this is the flow based on what I saw in Zeplin, but please correct me if I'm wrong about any of this.

## Code
It seems like the preferred interaction is: 
1.  User drags a file anywhere on the editor
1.  Dragging triggers file/file type detection, handling, and UI
1.  On file drop, file gets added to project folder

## Design
<img width="462" alt="screen shot 2017-06-27 at 10 54 11 am" src="https://user-images.githubusercontent.com/9204835/27602368-63a72a32-5b27-11e7-8695-354b84aff856.png">

The styling below is approximate and uses general variable names, units, font weights, and measurements, but the ultimate goal is to replicate the above design. Final styling should be in `em` units instead of `px`, color variables and font weights should use the existing SCSS names, etc.

## Window styling
```
 {
  box-shadow: inset -4px -4px 4px 0 rgba(0, 0, 0, 0.5);
  border: solid 4px var(--bluish);
}
```

### Background
```
{
  width: 271px;
  height: 240px;
  border-radius: 2px;
  background-color: var(--white-three);
  box-shadow: 0 12px 12px 0 rgba(0, 0, 0, 0.16);
  border: solid 0.5px #b9d0e1;
}
```
### Header
```
 {
  font-family: Montserrat;
  font-size: 21px;
  font-weight: bold;
  text-align: left;
  color: rgba(51, 51, 51, 0.87);
}
```
### Body
```
{
  font-family: Montserrat;
  font-size: 16px;
  line-height: 1.25;
  text-align: left;
  color: var(--black);
}
```

**Comment by machinic** [comment]
Sounds about right. I am traveling through Thursday evening. I can add some flow/state details on Friday as well as field any questions that might've come up in the thread.

**Comment by lmccart** [comment]
that looks right to me

**Comment by catarak** [comment]
i agree, this is confusing! i'd like to prioritize working on this, going to add it to the current milestone.

**Comment by ashu8912** [comment]
Yes The Asset Loading Can Take A Green Color Something that depicts the completion.
I am not a designer but the gray color is very confusing.

**Comment by pcgamer1** [comment]
I would like to work on this!

**Comment by ayushman-t** [comment]
Hi! I’m Ayushman Tripathi, JavaScript Software Developer and I came across the projects Asset Uploading Improvements, and Adding Test Coverage. I find them really fascinating. I believe my skills and experience would be a great fit for these project. I am really interested in contributing to Asset Uploading Improvements, and I’ve relevant skills for contributing to it as I’ve experience in full stack web development (React, Node, MongoDB, PostgresSQL, REST API, Django, AWS). I’ve worked on 10+ full stack projects & 4 internships in the past 1.5 years, and I have also completed the App Academy Open Software Engineering Bootcamp (500+ hours of full-stack online curriculum), where I learned a variety of skills. Additionally, I have worked on various frontend and backend projects, and I’m also preparing for the AWS Certified Developer Associate certification. I’m willing to learn new skills while contributing to the project. I’ll put all my efforts into successfully contributing to the project.
Can you please guide me on how to start contributing to the above two projects ?
Thanks a lot.

**Comment by ShivamAhir** [comment]
Hi, my name is **Shivam**. I'm currently a final year engineering **student at NIT Kurukshetra**. I have a strong understanding of front-end technologies such as _HTML, CSS, JavaScript, React.js, and Material UI_, and for backend development, I'm proficient in _Node.js_ and _Express.js_. I'm keen to contribute to this repository during the 2024 GSoC program. Could someone provide guidance on how to get started with this repository?

**Comment by rahulrana701** [comment]
@ShivamAhir first try to run the project locally and go through the code base thoroughly and then use that project as user for a while and then you can look at the issues and try to solve them , or can create issues yourself.

**Comment by CodeWPearl** [comment]
Hi! @catarak @machinic @lmccart 
I’d love to work on this issue if it’s still open. I’ve reviewed the discussion, related issues (#92, #271), and the proposed drag-and-drop flow, and I’m happy to help with implementing or refining the UI/UX and handling the file upload logic.

Please let me know if I can be assigned or if there are any specific requirements or next steps I should follow. Thanks!
